File: api/enums/document.py
from enum import IntEnum


class UserDocumentAuthority(IntEnum):
    OWNER = 0

class DocumentCategory(IntEnum):
    FILE = 0
    WEBSITE = 1
    QUICK_NOTE = 2
    AUDIO = 3

class DocumentMdConvertStatus(IntEnum):
    WAIT_TO = 0
    CONVERTING = 1
    SUCCESS = 2
    FAILED = 3

class DocumentEmbeddingStatus(IntEnum):
    WAIT_TO = 0
    EMBEDDING = 1
    SUCCESS = 2
    FAILED = 3

class DocumentGraphStatus(IntEnum):
    WAIT_TO = 0
    BUILDING = 1
    SUCCESS = 2
    FAILED = 3

class DocumentPodcastStatus(IntEnum):
    WAIT_TO = 0
    GENERATING = 1
    SUCCESS = 2
    FAILED = 3

class DocumentTranscribeStatus(IntEnum):
    WAIT_TO = 0
    TRANSCRIBING = 1
    SUCCESS = 2
    FAILED = 3

class DocumentSummarizeStatus(IntEnum):
    WAIT_TO = 0
    SUMMARIZING = 1
    SUCCESS = 2
    FAILED = 3

class DocumentProcessStatus(IntEnum):
    WAIT_TO = 0
    PROCESSING = 1
    SUCCESS = 2
    FAILED = 3

File: web/src/enums/document.ts
export enum DocumentCategory {
    FILE = 0,
    WEBSITE = 1,
    QUICK_NOTE = 2,
    AUDIO = 3
}

export enum DocumentMdConvertStatus {
    WAIT_TO = 0,
    CONVERTING = 1,
    SUCCESS = 2,
    FAILED = 3,
}

export enum DocumentEmbeddingStatus {
    WAIT_TO = 0,
    Embedding = 1,
    SUCCESS = 2,
    FAILED = 3,
}

export enum DocumentTranscribeStatus {
    WAIT_TO = 0,
    TRANSCRIBING = 1,
    SUCCESS = 2,
    FAILED = 3,
}

export enum DocumentPodcastStatus {
    WAIT_TO = 0,
    GENERATING = 1,
    SUCCESS = 2,
    FAILED = 3,
}

export enum DocumentProcessStatus {
    WAIT_TO = 0,
    PROCESSING = 1,
    SUCCESS = 2,
    FAILED = 3,
}

export enum DocumentGraphStatus {
    WAIT_TO = 0,
    BUILDING = 1,
    SUCCESS = 2,
    FAILED = 3
}

export enum DocumentSummarizeStatus {
    WAIT_TO = 0,
    SUMMARIZING = 1,
    SUCCESS = 2,
    FAILED = 3
}

export enum SectionDocumentIntegration {
    WAIT_TO = 0,
    SUPPLEMENTING = 1,
    SUCCESS = 2,
    FAILED = 3
}